lib/site_prism/element_container.rb

```diff
@@ -169,9 +177,9 @@ def create_existence_checker(element_name, *find_args)
         method_name = "has_#{element_name}?"
         create_helper_method(method_name, *find_args) do
           define_method(method_name) do |*runtime_args|
-            wait_time = SitePrism.use_implicit_waits ? max_wait_time : 0
-            visibility_args = { wait: wait_time }
-            element_exists?(*merge_args(find_args, runtime_args, **visibility_args))
+            visibility_args = { wait: checker_wait_time }
+            args = merge_args(find_args, runtime_args, **visibility_args)
```

Comment:
Curious, what's the double splat in all of these here for?

Reply:
This is to retain the exploded hash

i.e. (Ripped from SO)

```
def foo(a, *b, **c)
  [a, b, c]
end


> foo 10
=> [10, [], {}]
> foo 10, 20, 30
=> [10, [20, 30], {}]
> foo 10, 20, 30, d: 40, e: 50
=> [10, [20, 30], {:d=>40, :e=>50}]
> foo 10, d: 40, e: 50
=> [10, [], {:d=>40, :e=>50}]
```

Essentially we've had a huge long standing bug, where multiple keyword arguments were being ignored (We only ever tested with 0 or 1), Thomas from Capybara fixed this up in the merge_args method

Reply:
This is a little different though, isn't it?  This isn't using keyword args at all and `merge_args`'s declaration isn't the one doing the exploding.  The double-splat is in the method call, not declaration.  

```rb
def bar_args(find_args, runtime_args, override_options = {})
  [find_args, runtime_args, override_options]
end

# no splatter
> bar_args(10, 20, {a:1, b:2})
=> [10, 20, {:a=>1, :b=>2}]
# splatter
> bar_args(10, 20, **{a:1, b:2})
=> [10, 20, {:a=>1, :b=>2}]
> bar_args(10, 20, {a:1, b:2}) == bar_args(10, 20, **{a:1, b:2})
=> true
```

Reply:
I'll look into this. I know this "works" but I'll remove that and see what happens. It might have been something I put in for `spec` passing. But not 100%

Give me a few days to triage.